.pre-commit-config.yaml

```
@@ -1,23 +1,23 @@
 repos:
   - repo: https://github.com/psf/black
-    rev: 22.3.0
+    rev: 24.1.1
     hooks:
       - id: black-jupyter
         language_version: python3
 
   - repo: https://github.com/PyCQA/flake8
-    rev: 4.0.1
+    rev: 7.0.0
     hooks:
     -   id: flake8
         language_version: python3
 
   - repo: https://github.com/pycqa/isort
-    rev: 5.11.5
+    rev: 5.13.2
     hooks:
       -  id: isort
 
   - repo: https://github.com/asottile/pyupgrade
-    rev: v2.31.0
+    rev: v3.15.0
     hooks:
     -   id: pyupgrade
-        args: ['--py37-plus']
+        args: ['--py39-plus']
```

3.8 is our minimum python version
```suggestion
        args: ['--py38-plus']
```

---

No, all good. From your changes in this PR, it didn't update any other files.